### PR TITLE
Some typescript migration

### PR DESCRIPTION
**Pull request by joel-jeremy**
Some typescript migration for components in the root components directory. Also moved some sidebar components to the sidebar folder. There's no functional change in this PR. 

This is to help with [#1483](https://github.com/actualbudget/actual/issues/1483).

### CODE REVIEW

File: packages/desktop-client/src/components/sidebar/SidebarWithData.tsx [hunk 1]
```diff
@@ -63,10 +68,11 @@ function EditableBudgetName({ prefs, savePrefs }) {
           }}
           defaultValue={prefs.budgetName}
           onEnter={async e => {
-            const newBudgetName = e.target.value;
+            const inputEl = e.target as HTMLInputElement;
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/sidebar/SidebarWithData.tsx:71]: **❓ question:** ‏why do we need this manual type coercion? `e` should already be `KeyboardEvent<HTMLInputElement>`.

File: packages/desktop-client/src/components/App.tsx [hunk 2]
```diff
@@ -0,0 +1,181 @@
+import React, { useEffect, useState } from 'react';
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/App.tsx:1]: This file has quite a lot of functional changes (you're porting it from a class component to a hook). Any chance you could do this as a separate PR so we can carefully review the changes?

File: packages/desktop-client/src/components/FatalError.tsx [hunk 3]
```diff
@@ -0,0 +1,194 @@
+import React, { useState } from 'react';
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/FatalError.tsx:1]: Same comment about this component.

File: packages/desktop-client/src/components/sidebar/Accounts.tsx [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+  offBudgetAccountPath,
+  getBalanceQuery,
+  getAllAccountBalance,
+  getOnBudgetBalance,
+  getOffBudgetBalance,
+  showClosedAccounts,
+  onAddAccount,
+  onToggleClosedAccounts,
+  onReorder,
+}) {
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/sidebar/Accounts.tsx:27]: **💬 suggestion:** ‏we're missing the prop type here.
```suggestion
}: AccountsProps) {
```

File: packages/desktop-client/src/components/sidebar/Item.tsx [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+      style={{
+        flexDirection: 'row',
+        alignItems: 'center',
+        height: 20,
+      }}
+    >
+      <Icon width={15} height={15} />
+      <Block style={{ marginLeft: 8 }}>{title}</Block>
+      <View style={{ flex: 1 }} />
+      {Button && <Button />}
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/sidebar/Item.tsx:71]: **💬 suggestion:** ‏looks like this `Button` prop is unused and can be removed instead of refactoring it from `button` -> `Button`.

File: packages/desktop-client/src/components/sidebar/SecondaryItem.tsx [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+// eslint-disable-next-line no-restricted-imports
+import { styles, colors } from '../../style';
+import Block from '../common/Block';
+import View from '../common/View';
+
+import { ItemContent } from './ItemContent';
+
+const fontWeight = 600;
+
+let accountNameStyle = {
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/sidebar/SecondaryItem.tsx:17]: **💬 suggestion:** ‏we could import this from the `Account` component file to reduce duplication.

File: packages/desktop-client/src/components/sidebar/Sidebar.tsx [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+  getAllAccountBalance,
+  getOnBudgetBalance,
+  getOffBudgetBalance,
+  showClosedAccounts,
+  isFloating,
+  onFloat,
+  onAddAccount,
+  onToggleClosedAccounts,
+  onReorder,
+}) {
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/sidebar/Sidebar.tsx:37]: **💬 suggestion:** ‏missing prop type
```suggestion
}: SidebarProps) {
```

File: packages/desktop-client/src/components/App.tsx (hunk 2 above)
Comment (joel-jeremy) [reply packages/desktop-client/src/components/App.tsx:1]: Create a separate PR for this: https://github.com/actualbudget/actual/pull/1535

File: packages/desktop-client/src/components/sidebar/Accounts.tsx (hunk 4 above)
Comment (joel-jeremy) [reply packages/desktop-client/src/components/sidebar/Accounts.tsx]: Added the props :)

File: packages/desktop-client/src/components/sidebar/Item.tsx (hunk 5 above)
Comment (joel-jeremy) [reply packages/desktop-client/src/components/sidebar/Item.tsx]: Removed :)

**Comment by MatissJanis** [comment]
Apart from that one comment everything else looks good. :+1:

File: packages/desktop-client/src/components/sidebar/SidebarWithData.tsx (hunk 1 above)
Comment (MatissJanis) [reply packages/desktop-client/src/components/sidebar/SidebarWithData.tsx:72]: Ok, lets keep it as-is. Maybe we can fix it somehow later, but either way it's a nitpick.

File: packages/desktop-client/src/components/sidebar/Sidebar.tsx [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+import ToggleButton from './ToggleButton';
+import Tools from './Tools';
+
+import { useSidebar } from '.';
+
+export const SIDEBAR_WIDTH = 240;
+
+type SidebarProps = {
+  style: CSSProperties;
+  BudgetName: ComponentType;
```
[the remark this answers is not in the document]
Comment (joel-jeremy) [reply packages/desktop-client/src/components/sidebar/Sidebar.tsx]: Updated to ReactNode

**Review by MatissJanis — APPROVED** [review]
